### PR TITLE
chore: remove Garbage Collection ;)

### PR DESCRIPTION
**Pull request by dpc**
I'm a simple developer. I see a "GC" I find the way to get rid of it.

https://www.reddit.com/r/rustjerk/comments/n8b09i/ron_swanson_on_garbage_collection/

### CODE REVIEW

File: fedimint-client/src/lib.rs [hunk 1]
```diff
@@ -24,8 +24,9 @@
 //! interaction is required to drive them forward.
 //!
 //! ### State Machine Contexts
-//! State machines have access to both a [global context](sm::GlobalContext) as
-//! well as to a [module-specific context](module::ClientModule::context).
+//! State machines have access to both a [global
```
Comment (dpc) [inline fedimint-client/src/lib.rs:27]: Autoformatter insists. I kind of don't care.

File: fedimint-client/src/sm/executor.rs [hunk 2]
```diff
@@ -707,22 +698,22 @@ impl ExecutorBuilder {
 
 /// A state that is able to make progress eventually
 #[derive(Debug)]
-pub struct ActiveStateKey<GC> {
```
Comment (dpc) [inline fedimint-client/src/sm/executor.rs:710]: image

**Review by elsirion — APPROVED** [review]
You got a point, I probably overdid it a bit with the SM executor's generality …

Also, that meme made my day :rofl:

**Review by m1sterc001guy — APPROVED** [review]
Looks like it makes the code simpler to me

**Pull request merged**